monitor/pipeline_status.go

```diff
@@ -0,0 +1,30 @@
+package monitor
+
+import (
+	"sync"
+)
+
+var (
+	// pipelineStatusMap stores the latest pipeline status for each stream
+	pipelineStatusMap = make(map[string]PipelineStatus)
+	pipelineStatusMu  sync.RWMutex
+)
```

Comment:
WDYT of creating a struct to hold this "contiguous state"? So it's not N loose static vars, only 1 that holds everything and has the methods to interact with it safely (same functions below, just in the struct)

Reply:
Since this in memory store is not used yet, we can do it in a subsequent pr if you agree, since we want to merge this asap for evan